test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/RazorEngineTest.cs

```diff
@@ -0,0 +1,247 @@
+
+            var engine = RazorEngine.Create(b =>
+            {
+                InjectDirective.Register(b);
+                ModelDirective.Register(b);
+
+                b.AddTargetExtension(new InjectDirectiveTargetExtension());
+
+                b.Features.Add(new ModelExpressionPass());
+                b.Features.Add(new MvcViewDocumentClassifierPass());
+                b.Features.Add(new DefaultInstrumentationPass());
+
+                b.Features.Add(new DefaultTagHelperFeature());
+                b.Features.Add(GetMetadataReferenceFeature());
+            });
+
+            var inputContent = ResourceFile.ReadResource(_assembly, inputFile, sourceFile: true);
+            var fileProvider = new TestFileProvider();
+            fileProvider.AddFile(inputFile, inputContent);
+            var fileInfo = fileProvider.GetFileInfo(inputFile);
+            var razorTemplateEngine = new MvcRazorTemplateEngine(engine, new DefaultRazorProject(fileProvider));
+            var razorProjectItem = new DefaultRazorProjectItem(fileInfo, basePath: null, path: inputFile);
+
+            // Act
+            var codeDocument = razorTemplateEngine.CreateCodeDocument(razorProjectItem);
+            codeDocument.Items["SuppressUniqueIds"] = "test";
+
+            var csharpDocument = razorTemplateEngine.GenerateCode(codeDocument);
+
+            // Assert
+            Assert.Empty(csharpDocument.Diagnostics);
+
+#if GENERATE_BASELINES
+            ResourceFile.UpdateFile(_assembly, outputFile, expectedCode, csharpDocument.GeneratedCode);
+#else
+            Assert.Equal(expectedCode, csharpDocument.GeneratedCode, ignoreLineEndingDifferences: true);
+#endif
+        }
+
+        private static void RunDesignTimeTest(string testName)
+        {
+            // Arrange
+            var inputFile = "TestFiles/Input/" + testName + ".cshtml";
+            var outputFile = "TestFiles/Output/DesignTime/" + testName + ".cs";
+            var expectedCode = ResourceFile.ReadResource(_assembly, outputFile, sourceFile: false);
+
+            var lineMappingOutputFile = "TestFiles/Output/DesignTime/" + testName + ".mappings.txt";
+            var expectedMappings = ResourceFile.ReadResource(_assembly, lineMappingOutputFile, sourceFile: false);
+
+            var engine = RazorEngine.CreateDesignTime(b =>
+            {
+                InjectDirective.Register(b);
+                ModelDirective.Register(b);
+
+                b.AddTargetExtension(new InjectDirectiveTargetExtension());
+
+                b.Features.Add(new ModelExpressionPass());
+                b.Features.Add(new MvcViewDocumentClassifierPass());
+
+                b.Features.Add(new DefaultTagHelperFeature());
+                b.Features.Add(GetMetadataReferenceFeature());
+            });
+
+            var inputContent = ResourceFile.ReadResource(_assembly, inputFile, sourceFile: true);
+            var fileProvider = new TestFileProvider();
```

Comment:
That is how it was initially. But that won't add any of the default imports that `MvcRazorTemplateEngine` adds